gemrb/plugins/SDLVideo/SDL20GLVideo.cpp

```diff
@@ -624,7 +630,7 @@ void GLVideoDriver::BlitGameSprite(const Sprite2D* spr, int x, int y, unsigned i
 	Region src(dx, dy, w, h);
 	Region dst(tx + dx, ty + dy, w, h);
 
-	if (tint.r == 0 && tint.g == 0 && tint.b == 0)
+	if (!(flags & BLIT_TINTED) || (tint.r == 0 && tint.g == 0 && tint.b == 0))
```

Comment:
There was no check for applying tinting by given bits at all yet. This "is it black?" also looks a little suspicious, but just not to break something unforeseen ..